extensions/2.0/Vendor/EXT_feature_metadata/examples/weather/weather.gltf

```diff
@@ -0,0 +1,219 @@
+              "valueType": "FLOAT32",
+              "elementType": "ARRAY",
+              "valueCount": 3
+            }
+          }
+        },
+        "town": {
+          "name": "Town Data",
+          "properties": {
+            "name": {
+              "name": "Name",
+              "valueType": "STRING"
+            },
+            "population": {
+              "name": "Population",
+              "valueType": "FLOAT32"
+            }
+          }
+        }
+      },
+      "featureCollections": {
+        "weatherCollection": {
+          "featureClass": "weather",
+          "featureCount": 1000,
+          "properties": {
+            "airTemperature": {
+              "elementCount": 1000,
+              "bufferView": 2
+            },
+            "airPressure": {
+              "elementCount": 1000,
+              "bufferView": 3
+            },
+            "windVelocity": {
+              "elementCount": 1000,
+              "bufferView": 4
+            }
+          }
+        },
+        "townCollection": {
+          "featureClass": "town",
+          "featureCount": 3,
+          "properties": {
+            "name": {
+              "elementCount": 3,
+              "bufferView": 5,
+              "offsetsBufferView": 6
+            },
+            "population": {
+              "elementCount": 3,
+              "bufferView": 7
+            }
+          }
+        }
+      }
+    }
+  },
+  "scene": 0,
+  "scenes": [
+    {
+      "name": "Scene",
+      "nodes": [
+        0
+      ]
+    }
+  ],
+  "nodes": [
+    {
+      "name": "Point Cloud",
+      "mesh": 0
+    }
+  ],
+  "meshes": [
+    {
+      "name": "Point Cloud Mesh",
+      "primitives": [
+        {
+          "attributes": {
+            "POSITION": 0,
+            "_FEATURE_ID_0": 1
+          },
+          "mode": 0,
+          "extensions": {
+            "EXT_feature_metadata": {
+              "featureMappings": [
+                {
+                  "featureCollection": "weatherCollection",
+                  "featureIds": {
+                    "constant": 0,
+                    "vertexStride": 1
+                  }
+                },
+                {
+                  "featureCollection": "townCollection",
+                  "featureIds": {
+                    "attribute": "_FEATURE_ID_0"
+                  }
+                }
+              ]
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "accessors": [
+    {
+      "name": "Positions",
+      "bufferView": 0,
+      "byteOffset": 0,
+      "componentType": 5126,
+      "count": 1000,
+      "max": [
+        0.9984954469408285,
+        0.9919821104589452,
+        0.9970798239119962
+      ],
+      "min": [
+        -0.9968233643517341,
+        -0.9973561515575708,
+        -0.9991830031102058
+      ],
+      "type": "VEC3"
+    },
+    {
+      "name": "Town Feature ID",
+      "bufferView": 1,
+      "byteOffset": 0,
+      "componentType": 5121,
```

Comment:
Dragging the glTF model into [glTF-Validator](https://github.khronos.org/glTF-Validator/) has this error message:

> Vertex attribute data must be aligned to 4-byte boundaries

Basically, every element in an attribute needs to be aligned to 4 bytes. And glTF does not allow UNSIGNED_INT for vertex attribute accessors. Which leaves little choice but to use FLOAT.

Reply:
@lilleyse What does this mean for the spec? For meshes with feature IDs, this limits you to `2^24` vertices, beyond that you run the risk of rounding feature IDs incorrectly [see here](https://stackoverflow.com/questions/3793838/which-is-the-first-integer-that-an-ieee-754-float-is-incapable-of-representing-e#:~:text=The%20largest%20value%20representable%20by,However.)

Is there a reason why glTF limits the use of `UNSIGNED_INT` for accessors? Is there a restriction in WebGL or something?

Reply:
I did update the `weather` example to use floats. no problem here since I only defined 3 different possible features. but overall this feels a bit limiting.

Reply:
[vertexAttribPointer](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/vertexAttribPointer) doesn't support unsigned int, not even in WebGL 2